### PR TITLE
[#1744] ignore test on Java 16 and later

### PR DESCRIPTION
**Pull request by cdietrich**
[#1744] ignore test on Java 16 and later
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/ManifestMergerTest.java [hunk 1]
```diff
@@ -16,20 +16,64 @@
...
+		
+		@Override
+		public Statement apply(Statement base, FrameworkMethod method, Object target) {
+			if (isJava16OrLater) {
+				return new Statement() {
+					
+					@Override
+					public void evaluate() throws Throwable {
+						// do nothing
+						Assume.assumeTrue("Ignored on Java 16 and later", false);
```
Comment (cdietrich) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/ManifestMergerTest.java:70]: am not sure if we can do something to show up in test report as ignored

**Comment by cdietrich** [comment]
https://ci.eclipse.org/xtext/job/xtext-core/job/cd_issue1744/

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/ManifestMergerTest.java [hunk 2]
```diff
@@ -16,20 +16,64 @@
...
+		}
+		return false;
+	}
+	
+	@Rule
+	public MethodRule ignoreOnJava16OrLater = new MethodRule() {
+		
+		@Override
+		public Statement apply(Statement base, FrameworkMethod method, Object target) {
+			if (isJava16OrLater) {
```
Comment (szarnekow) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/ManifestMergerTest.java:64]: Maybe 
```
  return new Statement() {
     @Override
					public void evaluate() throws Throwable {
						// do nothing
						Assume.assumeFalse("Ignored on Java 16 and later", isJava16OrLater);
						base.evaluate();                
					} 
  };
```

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/ManifestMergerTest.java [hunk 3]
```diff
@@ -16,20 +16,64 @@
...
+				return version >= 16;
+			}
+		} catch (NumberFormatException e) {
+			// ok
+		}
+		return false;
+	}
+	
+	@Rule
+	public MethodRule ignoreOnJava16OrLater = new MethodRule() {
```
Comment (szarnekow) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/ManifestMergerTest.java:60]: I like the idea a lot. JUnit5 has something like `@EnabledOnJre(..)` / `@DisabledOnJre` respectively. We could try to mimic that (if we need similar test-toggles in other classes, too).

**Review by szarnekow — APPROVED** [review]

**Comment by szarnekow** [comment]
> https://ci.eclipse.org/xtext/job/xtext-core/job/cd_issue1744/

https://ci.eclipse.org/xtext/job/xtext-core/job/cd_issue1744/6/testReport/org.eclipse.xtext.generator/ManifestMergerTest/

**Pull request merged**